src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs

```diff
@@ -254,11 +254,55 @@ protected virtual void WriteChildResourceEntries()
                     {
                         WriteResourceCollectionEntry(resource.ResourceCollection, signature);
                     }
+
+                    if (!(This is MgmtExtensionClient)) // we don't need to generate `Get{Resource}` methods in ExtensionClient
+                    {
+                        WriteChildResourceGetMethod(resource.ResourceCollection, true);
+                        WriteChildResourceGetMethod(resource.ResourceCollection, false);
+                    }
                 }
             }
             _writer.Line();
         }
 
+        protected void WriteChildResourceGetMethod(ResourceCollection resourceCollection, bool isAsync)
+        {
+            var getOperation = resourceCollection.GetOperation;
+            // Copy the original method signature with changes in name and modifier (e.g. when adding into extension class, the modifier should be static)
+            var methodSignature = getOperation.MethodSignature with
+            {
+                // name after `Get{ResourceType}`
+                Name = $"{getOperation.MethodSignature.Name}{resourceCollection.Resource.Type.Name}",
+                Modifiers = GetMethodModifiers(),
+                // There could be parameters to get resource collection
+                Parameters = GetParametersForCollectionEntry(resourceCollection).Concat(GetParametersForResourceEntry(resourceCollection)).ToArray(),
+            };
+
+            using (WriteCommonMethodCommentsAndSignature(methodSignature, getOperation.ReturnsDescription != null ? getOperation.ReturnsDescription(isAsync) : null, isAsync))
```

Comment:
We might need a `writer.Line();` before this so that we do not have all the method glued together.